### PR TITLE
[Docs] clean up deployment guides

### PR DESCRIPTION
**Pull request by yubozhao**
Add install instruction for `jq` and avoid use `we`

### CODE REVIEW

File: docs/source/deployment/knative.rst [hunk 1]
```diff
@@ -26,16 +26,7 @@ Prerequisites
...
-prediction services on Knative with the docker image.
-
-===========================
-Build and push docker image
-===========================
-
-First, we need to build and push our BentoService to a docker registry.
-We will use the IrisClassifier BentoService from the getting
-:doc:`Quick start guide<../quickstart>`.
+This guide use the IrisClassifier BentoService from the :doc:`Quick start guide<../quickstart>`.
```
Comment (parano) [inline docs/source/deployment/knative.rst:29]: remove the "Saved BentoService bundle" in Prerequisites, and add "pip install bentoml scikit-learn" to Prerequisites?

File: docs/source/deployment/knative.rst [hunk 2]
```diff
@@ -178,7 +197,7 @@ For this guide, our kubernetes cluster run on minikube, we will get the appropri
     31871
 
 
-With the ip address and port, we can make a curl request to the prediction result from Knative
+With the ip address and port, makes a curl request to the prediction result from Knative
```
Comment (parano) [inline docs/source/deployment/knative.rst:200]: nit: With the IP address and port, Use `curl` to make an HTTP request to the deployment in Knative:

File: docs/source/deployment/knative.rst (hunk 1 above)
Comment (parano) [reply docs/source/deployment/knative.rst:29]: can you add minikube setup link to the `Kubernetes cluster`prerequisite

File: docs/source/deployment/knative.rst [hunk 3]
```diff
@@ -153,7 +171,7 @@ Create bentoml namespace and then deploy BentoService to Knative with kubectl ap
 
 
 
-We can monitor the status with kubectl get ksvc command.
+Monitor the status with kubectl get ksvc command.
```
Comment (parano) [inline docs/source/deployment/knative.rst:174]: View the status of the deployment with `kubectl get ksvc` command:

File: docs/source/deployment/knative.rst (hunk 1 above)
Comment (parano) [reply docs/source/deployment/knative.rst:29]: Add something like "This guide demonstrates how to deploy a machine learning model to KNative using bentoml ..." to the end of "Deploying to Knative" section at top of this file?

File: docs/source/deployment/knative.rst [hunk 4]
```diff
@@ -167,7 +185,8 @@ Validate prediction server with sample data
 ===========================================
 
 
-For this guide, our kubernetes cluster run on minikube, we will get the appropriate ip from minikube and the port from istio
+For this guide, the kubernetes cluster run on minikube, Get the appropriate ip from
```
Comment (parano) [inline docs/source/deployment/knative.rst:188]: ~For this guide~, ~the appropriate ip~

Mention `minikube` setup at the `Prerequisites` section

change this to:

Find the IP address and exposed port of the deployed Knative service:

File: docs/source/deployment/google_cloud_run.rst [hunk 5]
```diff
@@ -67,7 +67,7 @@ This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide
 
 .. code-block:: bash
 
-    > bentoml get IrisClassifier:latest
```
Comment (parano) [inline docs/source/deployment/google_cloud_run.rst:70]: why removing `>`, other bash commands all have this right?

File: docs/source/deployment/google_cloud_run.rst [hunk 6]
```diff
@@ -103,13 +103,33 @@ This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide
     }
 
 
+After saving the BentoService instance, you can now start a REST API server with the
+model trained and test the API server locally:
```
Comment (parano) [inline docs/source/deployment/google_cloud_run.rst:107]: The BentoML saved bundle created can now be used to start a REST API Server hosting the BentoService and available for sending test request:

File: docs/source/deployment/google_cloud_run.rst [hunk 7]
```diff
@@ -67,7 +67,7 @@ This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide
 
 .. code-block:: bash
```
Comment (parano) [inline docs/source/deployment/google_cloud_run.rst:68]: Add a little bit more context to `This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`:` ?

File: docs/source/deployment/knative.rst (hunk 1 above)
Comment (parano) [reply docs/source/deployment/knative.rst:29]: actually something like what we had for kubeflow guide would be great, just modify based on this:

```
This guide demonstrates how to serve a scikit-learn based iris classifier model with BentoML on a Kubernetes cluster. The same deployment steps are also applicable for models trained with other machine learning frameworks, see more BentoML examples here.
```

File: docs/source/deployment/knative.rst [hunk 8]
```diff
@@ -81,22 +72,49 @@ Use BentoML CLI tool to get the information about IrisClassifier.
     }
 
 
-Navigate to the BentoService archive bundle location. Build and push docker image to docker hub.
+After saving the BentoService instance, you can now start a REST API server with the
```
Comment (parano) [inline docs/source/deployment/knative.rst:75]: same as the other comment

File: docs/source/deployment/knative.rst [hunk 9]
```diff
@@ -81,22 +72,49 @@ Use BentoML CLI tool to get the information about IrisClassifier.
...
+Deploy BentoService to Knative
+==============================
 
-=================
-Deploy to Knative
-=================
+BentoML provides a convenient way of containerizing the model API server with Docker. To
+create a docker container image for the sample model above:
 
+1. Find the file directory of the SavedBundle with bentoml get command, which is directory
```
Comment (parano) [inline docs/source/deployment/knative.rst:101]: can you use the version I rewrote in README.md?

Comment (parano) [reply docs/source/deployment/knative.rst:101]: * Find the SavedBundle directory with bentoml get command

* Run docker build with the SavedBundle directory which contains a generated Dockerfile

* Run the generated docker image to start a docker container serving the model

File: docs/source/deployment/knative.rst [hunk 10]
```diff
@@ -81,22 +72,49 @@ Use BentoML CLI tool to get the information about IrisClassifier.
...
+structured as a docker build context.
 
-Make sure Knative serving component and its pods are running.
+2. Running docker build with this directory produces a docker image containing the model
```
Comment (parano) [inline docs/source/deployment/knative.rst:104]: ~Running~ Run

Comment (parano) [reply docs/source/deployment/knative.rst:104]: ..built with..

Comment (parano) [reply docs/source/deployment/knative.rst:104]: nvm, see the comment above

File: docs/source/deployment/google_cloud_run.rst (hunk 5 above)
Comment (yubozhao) [reply docs/source/deployment/google_cloud_run.rst:70]: I think it is easier for user to copy.

File: docs/source/deployment/aws_ecs.rst [hunk 11]
```diff
@@ -36,11 +40,12 @@ Prerequisites
 AWS ECS deployment with BentoML
 -------------------------------------------------
 
-This guide will walk through from deploying BentoService with ECS, validate result with
-sample data and removing service and clean up AWS resources.
+This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`.
+The IrisClassifier has an endpoint, `/predict`, as its entry point for accessing the prediction
+service. The predict endpoint expects `pandas.DataFrame` as input.
```
Comment (parano) [inline docs/source/deployment/aws_ecs.rst:45]: "The prediction endpoint" or "The endpoint"

File: docs/source/deployment/aws_ecs.rst [hunk 12]
```diff
@@ -36,11 +40,12 @@ Prerequisites
 AWS ECS deployment with BentoML
 -------------------------------------------------
 
-This guide will walk through from deploying BentoService with ECS, validate result with
-sample data and removing service and clean up AWS resources.
+This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`.
+The IrisClassifier has an endpoint, `/predict`, as its entry point for accessing the prediction
```
Comment (parano) [inline docs/source/deployment/aws_ecs.rst:44]: let's just remove line 44 and 45

File: docs/source/deployment/aws_ecs.rst [hunk 13]
```diff
@@ -86,8 +91,26 @@ Use the IrisClassifier BentoService from the :doc:`Quick start guide<../quicksta
...
+    # Send test request:
+    curl -i \
+      --header "Content-Type: application/json" \
+      --request POST \
+      --data '[[5.1, 3.5, 1.4, 0.2]]' \
+      http://localhost:5000/predict
+
 ==================================================================
-Build and push docker image to AWS ECR(Elastic Container Registry)
+Build BentoML model server for AWS ECR(Elastic Container Registry)
```
Comment (parano) [inline docs/source/deployment/aws_ecs.rst:113]: change title to: Dockerize BentoML model server for deployment

And explain to create ECS deployment, it needs to build docker image and push the image to ECR.

Comment (parano) [reply docs/source/deployment/aws_ecs.rst:113]: can you also update the dockeriation steps below to the new one we've been using with `bentoml get ... | jq ...`

File: docs/source/deployment/azure_container_instance.rst [hunk 14]
```diff
@@ -171,14 +199,14 @@ Create and configure Azure ACR (Azure Container Registry)
...
 
-    > az acr login --name bentomlirisclassifier
+    az acr login --name bentomlirisclassifier
 
     Login Succeeded
 
 
 .. code-block:: bash
 
-    > az acr show --name BentoMLIrisClassifier --query loginServer --output table
```
Comment (parano) [inline docs/source/deployment/azure_container_instance.rst:181]: I think it's actually better to keep ">" or "$" here

It looks really bad now - it's the same format as the stdout result below now

File: docs/source/deployment/aws_ecs.rst [hunk 15]
```diff
@@ -36,11 +40,10 @@ Prerequisites
 AWS ECS deployment with BentoML
 -------------------------------------------------
 
-This guide will walk through from deploying BentoService with ECS, validate result with
-sample data and removing service and clean up AWS resources.
+This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`.
 
+Build the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`.
```
Comment (parano) [inline docs/source/deployment/aws_ecs.rst:45]: remove l45, it's repetitive to l43

File: docs/source/deployment/aws_ecs.rst [hunk 16]
```diff
@@ -86,32 +89,58 @@ Use the IrisClassifier BentoService from the :doc:`Quick start guide<../quicksta
...
+
+    # Send test request:
+    curl -i \
+      --header "Content-Type: application/json" \
+      --request POST \
+      --data '[[5.1, 3.5, 1.4, 0.2]]' \
+      http://localhost:5000/predict
+
+======================================================================
+Dockerize BentoML model server for AWS ECR(Elastic Container Registry)
```
Comment (parano) [inline docs/source/deployment/aws_ecs.rst:111]: Title: Dockerize BentoML model server for deployment

explain what is AWS ECR and why it's needed in the content below

File: docs/source/deployment/aws_ecs.rst [hunk 17]
```diff
@@ -130,8 +159,11 @@ Create AWS ECR repository
 
 .. code-block:: bash
 
-    > cd /Users/bozhaoyu/bentoml/repository/IrisClassifier/20200121141808_FE78B5
-    > docker build . --tag=192023623294.dkr.ecr.us-west-2.amazonaws.com/irisclassifier-ecs
+    # Download and install jq, the JSON processor: https://stedolan.github.io/jq/download/
```
Comment (parano) [inline docs/source/deployment/aws_ecs.rst:162]: `JSOn processor` is a bit inaccurate, let's use `command-line JSON processor`:

# Install jq, the command-line JSON processor: https://stedolan.github.io/jq/download/

File: docs/source/deployment/azure_container_instance.rst [hunk 18]
```diff
@@ -28,7 +32,12 @@ Prerequisites
 Deploying BentoService to Azure Container Instance
 --------------------------------------------------
 
-This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`:
+This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`.
+The IrisClassifier has an endpoint, `/predict`, as its entry point for accessing the prediction
```
Comment (parano) [inline docs/source/deployment/azure_container_instance.rst:36]: commented about removing this?

File: docs/source/deployment/azure_container_instance.rst [hunk 19]
```diff
@@ -28,7 +32,12 @@ Prerequisites
...
+service. The predict endpoint expects `pandas.DataFrame` as input.
+
+Build the IrisClassifier BentoService from the :doc:`quick start guide <../quickstart>`.
```
Comment (parano) [inline docs/source/deployment/azure_container_instance.rst:39]: again this is just repetitive.. rewrite this pls

File: docs/source/deployment/google_cloud_run.rst [hunk 20]
```diff
@@ -48,16 +53,21 @@ Create Google cloud project
...
 
-=============================================
-Build and push docker image to GCP repository
-=============================================
+============================================================
+Build and push BentoML model service image to GCP repository
+============================================================
+
+This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`.
+The IrisClassifier has an endpoint, `/predict`, as its entry point for accessing the prediction
```
Comment (parano) [inline docs/source/deployment/google_cloud_run.rst:66]: remove

File: docs/source/deployment/azure_container_instance.rst [hunk 21]
```diff
@@ -28,7 +32,12 @@ Prerequisites
 Deploying BentoService to Azure Container Instance
 --------------------------------------------------
 
-This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`:
+This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`.
```
Comment (parano) [inline docs/source/deployment/azure_container_instance.rst:35]: The context was explained in the first paragraph,  here simply say:

Run the example project form the `quick start guide <../quickstart>` to create the BentoML saved bundle for deployment:

File: docs/source/deployment/azure_container_instance.rst [hunk 22]
```diff
@@ -38,7 +47,7 @@ This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide
 
 .. code-block:: bash
```
Comment (parano) [inline docs/source/deployment/azure_container_instance.rst:48]: Add a note here before `bentoml get`:

Verify the saved bundle created:

File: docs/source/deployment/azure_container_instance.rst (hunk 21 above)
Comment (parano) [reply docs/source/deployment/azure_container_instance.rst:35]: can you do the same for the other docs too?

File: docs/source/deployment/azure_container_instance.rst (hunk 22 above)
Comment (parano) [reply docs/source/deployment/azure_container_instance.rst:48]: can you do the same for the other docs too?

File: docs/source/deployment/aws_ecs.rst (hunk 15 above)
Comment (parano) [reply docs/source/deployment/aws_ecs.rst:45]: see the other comment

File: docs/source/deployment/azure_container_instance.rst [hunk 23]
```diff
@@ -421,10 +454,10 @@ Validate Azure container instance with sample data POST request
...
 
-    > curl -X \
-      POST "http://bentomlirisclassifier777.eastus.azurecontainer.io:5000/predict" \
-      --header "Content-Type: application/json" \
-      -d '[[5.1, 3.5, 1.4, 0.2]]'
+    $ curl -X \
+        POST "http://bentomlirisclassifier777.eastus.azurecontainer.io:5000/predict" \
```
Comment (parano) [inline docs/source/deployment/azure_container_instance.rst:458]: could this line be 
```
POST $(az container show --resource-group iris-classifier --name bentomlirisclassifier --query ipAddress.fqdn)
```
?

no need to change now.. maybe for another time when you get a chance to verify if it works

File: docs/source/deployment/knative.rst [hunk 24]
```diff
@@ -1,41 +1,44 @@
...
 
+This guide demonstrates how to serve a scikit-learn based iris classifier model with
+BentoML on a KNative cluster. The same deployment steps are also applicable for models
+trained with other machine learning frameworks, see more BentoML examples :doc:`here <../examples>`.
 
 Prerequisites
 -------------
 
-* kubernetes cluster version 1.15 or newer.
+* A kubernetes installed cluster. `minikube` is used in this guide.
```
Comment (parano) [inline docs/source/deployment/knative.rst:15]: * A Kubernetes cluster
    * `minikube` is the recommended way to run Kubernetes locally: https://kubernetes.io/docs/tasks/tools/install-minikube/
    * Kubernetes guide: https://kubernetes.io/docs/setup/

File: docs/source/deployment/knative.rst [hunk 25]
```diff
@@ -1,41 +1,44 @@
...
 -------------------------------
 
-In this guide, we will build BentoService with docker and then deploy a
-prediction services on Knative with the docker image.
-
-===========================
-Build and push docker image
-===========================
+This guide uses the IrisClassifier BentoService from the :doc:`Quick start guide <../quickstart>`.
+The IrisClassifier has an endpoint, `/predict`, as its entry point for accessing the prediction
```
Comment (parano) [inline docs/source/deployment/knative.rst:38]: update according to the previous comment

File: docs/source/deployment/azure_container_instance.rst (hunk 23 above)
Comment (yubozhao) [reply docs/source/deployment/azure_container_instance.rst:458]: yeah, maybe next time when we have a chance verify it

**Pull request merged**